sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -59,6 +59,12 @@ def __init__(self, http_client, properties, ipython_display,
         super(LivySession, self).__init__()
         assert u"kind" in list(properties.keys())
         kind = properties[u"kind"]
+
+        if not should_heartbeat:
+            properties.pop(u"heartbeatTimeoutInSecond")
```

Comment:
On second thoughts, we should have all the logic for `heartbeatTimeoutInSecond` in the same place, and not in two different places. Other people will also miss it.